packages/backend/src/analytics/LightdashAnalytics.ts

```diff
@@ -165,8 +168,23 @@ export enum QueryExecutionContext {
     SCHEDULED_CHART = 'scheduledChart',
     SCHEDULED_DASHBOARD = 'scheduledDashboard',
     CALCULATE_TOTAL = 'calculateTotal',
+    API = 'api',
+    CLI = 'cli',
 }
 
+export const getContextFromHeader = (req: Request) => {
+    const method = getRequestMethod(req.header(LightdashRequestMethodHeader));
+    switch (method) {
+        case RequestMethod.CLI:
+        case RequestMethod.CLI_CI:
+            return QueryExecutionContext.CLI;
+        case RequestMethod.UNKNOWN:
+            return QueryExecutionContext.API;
+        default:
+            return undefined;
```

Comment:
I'm wondering if this is actually the correct way to distinguish requests.

I don't think we usually set `UNKNOWN` as the request method. I've seen a couple of cases in the code, but I don't quite understand the reason behind it. I believe the `default:` use case is more suitable for API calls.

You can probably check for PAT as well. That, and having no request method would indicate an API call.

Reply:
ok, I've just tried the following: 

- I made a request to get results from explore: that made a request with `method WEB_APP`

![image](https://github.com/user-attachments/assets/e64f9594-86c2-4fd4-89bb-652296d50933)

- I copied the same Request as CURL; and triggered the same request without modifying, using the same header, it obviously uses the same `method WEB_APP`

- Then I removed the `  -H 'Lightdash-Request-Method: WEB_APP' ` header, this method was `method UNKNOWN`


So for me it seems fair to use `UNKNOWN` to identify API requests. 
But if you are still not sure, or you think this can lead to false positives,  we can close this PR and tell @TuringLovesDeathMetal  it is not possible to reliably identify external `API` requests.

Reply:
oh I see, I think that covers it then. approving the PR